_posts/en/newsletters/2019-05-14-newsletter.md

```diff
@@ -0,0 +1,648 @@
+  <th>Total</th>
+  <td>132</td>
+  <td>48.75</td>
+  <td>51.25</td>
+</tr>
+</table>
+
+Besides the change from ECDSA to Schnorr for the signature algorithm,
+there are a few important (but easy to implement) changes to the
+transaction digest---the hash that a signature commits to in order to
+prove the transaction is an authorized spend of a UTXO.
+
+Most notably, the hash used goes from the legacy protocol's
+double-SHA256 (SHA256d) to a single SHA256 operation.  The extra hashing
+previously used isn't believed to have provided any meaningful security.
+Additionally, the data hashed is prefixed with a value that's specific
+to this use of the hash function; this helps prevent problems like
+[CVE-2012-2459][] where a hash from one context can be used in another
+context.  The prefix tag is `SHA256(tag) || SHA256(tag)` where the tag
+in this case is the UTF-8 string "TapSighash" and `||` stands for
+concatenation.  Software that needs to create or verify a large number
+of signatures (such as active LN nodes or Bitcoin full nodes) can use a
+version of their SHA256 function that's been pre-initialized with the
+prefix tag so it doesn't need to repeat those operations for every
+signature verification.  Implementations that don't require maximal
+performance (such as ordinary wallets) can just implement the algorithm
+as described using their default SHA256 library function.
+
+There are also some changes to what data is included in the hash and how
+it is serialized.  This includes improvements that can help make wallets
+without access to verified UTXO entries (e.g. hardware wallets and
+[HSMs][]) more efficient because they don't need to retrieve as much
+data in order to ensure they're signing for correct UTXOs and amounts.
+
+Although that may sound like a significant number of changes, we suspect
+it's only an afternoon worth of work making minor serialization changes
+for any wallet that's already segwit compatible and which can gain
+access to a library like [libsecp256k1][] for generating
+bip-schnorr-compatible signatures.
+
+## Simple multisig spending using Taproot
+
+After single-sig UTXOs, the most common are simple multisig UTXOs.
+These are outputs that depend on a certain number of signatures from
+particular pubkeys but don't have any other conditions.  These are used
+both by individual users (e.g. requiring multiple devices to work
+together in a spend) and by multiple parties to a single transaction
+(e.g. escrows).
+
+There are two ways to perform simple multisig spending using Taproot,
+the most efficient of which is key and signature aggregation as
+described in this subsection.  We'll examine the other mechanism in the
+next subsection.
+
+For aggregation, two or more private keys are created and their pubkeys
+are derived.  The pubkeys are then combined into a single pubkey that's
+indistinguishable from any other Bitcoin pubkey.  This is used as the
+segwit witness program (e.g. bech32 address) as described in the
```

Comment:
@harding - I'd suggest dropping "(e.g. bech32 address)" unless you think it makes this sentence a lot easier to understand.